aya/src/maps/array/mod.rs

```diff
@@ -4,6 +4,17 @@ mod array;
 mod per_cpu_array;
 mod program_array;
 
-pub use array::Array;
+pub use array::*;
 pub use per_cpu_array::PerCpuArray;
 pub use program_array::ProgramArray;
+
+use crate::maps::{MapData, MapError};
+
+pub(crate) fn check_bounds(map: &MapData, index: u32) -> Result<(), MapError> {
+    let max_entries = map.obj.max_entries();
+    if index >= map.obj.max_entries() {
```

Comment:
You could use already existing `max_entries` variable instead

```suggestion
    if index >= max_entries {
```

Reply:
Ah good catch thanks!